ledger-core/vanilla/cachekit/cache_example.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package cachekit
+
+type Key = uint64
+type Value = string
+
+const zeroValue = ""
+
+/*
+	UintCache is an example/template implementation of a cache that uses the Core.
+
+	To create a new cache type - copy this file, rename struct, set Key and Value type aliases accordingly, then
+	set zeroValue constant in accordance with Value type.
+
+
+ */
+
+func NewUintCache(cs Strategy) *UintCache {
+	sc := &UintCache{
+		keys: map[Key]Index{},
+		values: [][]uintEntry { make([]uintEntry, cs.GetAllocationPage()) },
+	}
+	sc.core = NewCore(cs, sc.trimBatch)
+
+	return sc
+}
+
+// UintCache is an example/template implementation of a cache that uses the Core.
+type UintCache struct {
+	values [][]uintEntry
+	keys   map[Key]Index
+
+	core   Core
+}
+
+type uintEntry struct {
+	key Key
+	value Value
+}
+
+// Allocated returns a total number of cache entries allocated, but some of them may be unused.
+// NB! Cache can only grow.
+func (p *UintCache) Allocated() int {
+	return len(p.values) * cap(p.values[0])
+}
+
+// Occupied returns a number of added / available cache entries.
+func (p *UintCache) Occupied() int {
+	return len(p.keys)
+}
+
+// Put adds value with the given key.
+// If key was already added, then cached value remains unchanged and the function returns (false).
+// Access to the key is always updated.
+func (p *UintCache) Put(key Key, value Value) bool {
+	idx, ok := p.keys[key]
+	if ok {
+		p.core.Touch(idx)
+		return false
+	}
+	idx, _ = p.core.Add()
+	p.keys[key] = idx
+	p.putEntry(idx, uintEntry{key, value})
+	return true
+}
+
+// Replace adds or replaces value with the given key. If key was already added, then cached value is updated and the function returns (false).
+// Access to the key is always updated.
+func (p *UintCache) Replace(key Key, value Value) bool {
+	idx, ok := p.keys[key]
+	if ok {
+		p.core.Touch(idx)
+		p.putEntry(idx, uintEntry{key, value})
+		return false
+	}
+	idx, _ = p.core.Add()
+	p.keys[key] = idx
```

Comment:
looks like here should be `p.putEntry(idx, uintEntry{key, value})`

Reply:
thanks! fixed